persistence/src/main/java/cz/fi/muni/pa165/dao/VisitDaoImpl.java

```diff
@@ -80,23 +83,29 @@ public void delete(Visit visit) {
      * @return list of all visits for given forest
      */
     public List<Visit> findByForest(Forest forest) {
-
-        throw new UnsupportedOperationException();
+        try {
+            return em.createQuery("select v from Visit v where forest = :forest",
+                    Visit.class).setParameter("forest", forest).getResultList();
+        } catch (NoResultException nrf) {
+            return null;
```

Comment:
according to these docs no: http://docs.oracle.com/javaee/6/api/javax/persistence/TypedQuery.html#getResultList()